hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -2578,6 +2578,18 @@ func reconcileControlPlaneOperatorDeployment(
 						},
 						Resources: cpoResources,
 					},
+					{
+						Name:    "socks-proxy",
+						Image:   utilitiesImage,
+						Command: []string{"/usr/bin/control-plane-operator", "konnectivity-socks5-proxy", "--resolve-from-guest-cluster-dns=true", "--resolve-from-management-cluster-dns=true"},
+						Args:    []string{"run"},
+						Resources: corev1.ResourceRequirements{
+							Requests: corev1.ResourceList{
+								corev1.ResourceCPU:    resource.MustParse("10m"),
+								corev1.ResourceMemory: resource.MustParse("10Mi"),
+							},
+						},
+					},
 				},
```

Comment:
Accidental convenient side effects:
I think this should not need certs given it goes through https://github.com/openshift/hypershift/blob/main/konnectivity-socks5-proxy/main.go#L125. Otherwise we would have a chicken-egg issue to get the certs.

The kube client for dataplane dns resolution would pick the incluster config of the CPO and so trying to resolve actually via management svc network, fail and fallback https://github.com/openshift/hypershift/blob/main/konnectivity-socks5-proxy/main.go#L73